Servers/UI/OJS.Servers.Ui/ClientApp/src/components/modal/ContestModal.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from 'react';
+import Box from '@mui/material/Box';
+import Modal from '@mui/material/Modal';
+import Typography from '@mui/material/Typography';
+import isNil from 'lodash/isNil';
+
+import { ContestParticipationType } from '../../common/constants';
+import { IIndexContestsType } from '../../common/types';
+import { useAppUrls } from '../../hooks/use-app-urls';
+import Button, { ButtonSize, LinkButton } from '../guidelines/buttons/Button';
+
+import styles from './ContestModal.module.scss';
+
+interface IContestModalProps {
+    contest: IIndexContestsType;
+    openModal: boolean;
+}
+
+const ContestModal = ({ contest, openModal }: IContestModalProps) => {
+    const [ open, setOpen ] = useState<boolean>(false);
+    const { getParticipateInContestUrl } = useAppUrls();
+
+    useEffect(() => console.log('test'));
+
+    console.log(openModal);
+    console.log(contest);
+    if (isNil(contest)) {
+        return null;
+    }
+
+    return (
+        <div>
+            <Modal
+              open={open}
+              onClose={() => setOpen(false)}
+              aria-labelledby="modal-modal-title"
+              aria-describedby="modal-modal-description"
+            >
+                <Box className={styles.modal}>
+                    <Typography id="modal-modal-title" variant="h6" className={styles.modalHeading}>
+                        <>
+                            Starting now you will have
+                            {' '}
+                            {contest.duration}
+                            {' '}
+                            hours to complete the contest
+                            {' '}
+                            {contest.name}
+                            {' '}
+                            { contest.numberOfProblems}
+                            .
```

Comment:
Isn't something missing here between the two?

Reply:
Fixed.